setuptools_rust/build.py

```diff
@@ -285,7 +285,14 @@ def build_extension(self, ext):
 
                 ext.install_script(ext_path)
             else:
-                ext_path = build_ext.get_ext_fullpath(target_fname)
+                # Technically it's supposed to contain a
+                # `setuptools.Extension`, but in practice the only attribute it
+                # checks is `ext.py_limited_api`.
+                build_ext.ext_map[ext.name] = ext
+                try:
+                    ext_path = build_ext.get_ext_fullpath(target_fname)
+                finally:
+                    del build_ext.ext_map[ext.name]
```

Comment:
Sneaky but I think that's fine to me ;)

Reply:
Actually maybe worth adding `assert ext.name not in build_ext.ext_map` before we start fiddling with this?

Reply:
Done!